### PR TITLE
Modify rule S116: Migrate to LayC - Field names should comply with a naming convention

### PR DESCRIPTION
**Pull request by ADarko22**

### CODE REVIEW

File: rules/S116/csharp/rule.adoc [hunk 1]
```diff
@@ -2,26 +2,37 @@
...
 class MyClass {
    private int myField;
 }
 ----
 
+== Resources
+
+=== Documentation
+
+* Microsoft - https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/naming-guidelines[.NET Naming Guidelines]
```
Comment (ADarko22) [inline rules/S116/csharp/rule.adoc:33]: is this reference relevant?

File: rules/S116/flex/rule.adoc [hunk 2]
```diff
@@ -2,26 +2,36 @@
...
 public class MyClass {
    public var myField:int;
 }
 ----
 
+== Resources
+
+=== Documentation
+
+* Wikipedia - https://en.wikipedia.org/wiki/Naming_convention_(programming)[Naming Convention (programming)]
```
Comment (ADarko22) [inline rules/S116/flex/rule.adoc:33]: could not find any specific documentation for FLEX

File: rules/S116/php/rule.adoc [hunk 3]
```diff
@@ -2,26 +2,37 @@
...
 class MyClass {
   $myField;
 }
 ----
 
+== Resources
+
+=== Documentation
+
+* php.net - https://www.php.net/manual/en/userlandnaming.rules.php[Userland Naming Guide]
```
Comment (ADarko22) [inline rules/S116/php/rule.adoc:33]: is this reference valid?

File: rules/S116/swift/rule.adoc [hunk 4]
```diff
@@ -2,26 +2,44 @@
...
 class MyClass {
   var myField = 1
 }
 ----
 
+== Resources
+
+=== Documentation
+
+* Swift - https://www.swift.org/documentation/api-design-guidelines/#naming[API Design Guidelines - Naming]
```
Comment (ADarko22) [inline rules/S116/swift/rule.adoc:33]: are the references valid?

File: rules/S116/cfamily/rule.adoc [hunk 5]
```diff
@@ -2,26 +2,37 @@
 
 include::../description.adoc[]
 
-=== Noncompliant code example
+== How to fix it
```
Comment (frederic-tingaud-sonarsource) [inline rules/S116/cfamily/rule.adoc:5]: The description is short and the code examples are super straightforward. I think just having the code samples in "Why is this an issue?" would be simpler.

File: rules/S116/cfamily/rule.adoc [hunk 6]
```diff
@@ -2,26 +2,37 @@
...
 
-[source,cpp]
+The examples below enforce the following regular expression ``++^[a-z][a-zA-Z0-9]*$++``:
+
+==== Noncompliant code example
+
+[source,cpp,diff-id=1,diff-type=noncompliant]
 ----
 class MyClass {
   int my_field;
```
Comment (frederic-tingaud-sonarsource) [inline rules/S116/cfamily/rule.adoc:16]:
```suggestion
  int my_field; // Noncompliant
```

File: rules/S116/cfamily/rule.adoc [hunk 7]
```diff
@@ -2,26 +2,37 @@
...
 ----
 
-=== Compliant solution
+==== Compliant solution
 
-[source,cpp]
+[source,cpp,diff-id=1,diff-type=compliant]
 ----
 class MyClass {
   int myField;
```
Comment (frederic-tingaud-sonarsource) [inline rules/S116/cfamily/rule.adoc:25]:
```suggestion
  int myField;  // Compliant
```

**Review by frederic-tingaud-sonarsource — CHANGES_REQUESTED** [review]
What would you think about having the code samples in the first tab? Given how straightforward this all is, I think that would work better.

File: rules/S116/csharp/rule.adoc (hunk 1 above)
Comment (renaud-tognelli-sonarsource) [reply rules/S116/csharp/rule.adoc:33]: It gives weight to the rule.
Maybe point it directly to the field name convention, wdyt?
https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/names-of-type-members#names-of-fields

File: rules/S116/php/rule.adoc (hunk 3 above)
Comment (renaud-tognelli-sonarsource) [reply rules/S116/php/rule.adoc:33]: There is not much about fields here, it is a bit misleading I would say.

File: rules/S116/swift/rule.adoc (hunk 4 above)
Comment (renaud-tognelli-sonarsource) [reply rules/S116/swift/rule.adoc:33]: same here

File: rules/S116/php/rule.adoc (hunk 3 above)
Comment (ADarko22) [reply rules/S116/php/rule.adoc:33]: True, but it still brings some value to an user who wants to dig into the topic

File: rules/S116/cfamily/rule.adoc [hunk 8]
```diff
@@ -2,26 +2,35 @@
 
 include::../description.adoc[]
 
-=== Noncompliant code example
+=== Code examples
```
Comment (frederic-tingaud-sonarsource) [inline rules/S116/cfamily/rule.adoc:5]: If we put the code in "Why is this an issue", we don't need the subtitles.

```
include::../description.adoc[]

For example with the following regular expression ``++^[a-z][a-zA-Z0-9]*$++``:

[source,cpp,diff-id=1,diff-type=noncompliant]
----
class MyClass {
  int my_field; // Noncompliant
};
----

Should be replaced with:

[source,cpp,diff-id=1,diff-type=compliant]
----
class MyClass {
  int myField;  // Compliant
};
----
```

**Review by frederic-tingaud-sonarsource — CHANGES_REQUESTED** [review]

**Review by frederic-tingaud-sonarsource — APPROVED** [review]

**Pull request merged**